### PR TITLE
CLN: f-string formatting

### PR DESCRIPTION
**Pull request by monicaw218**
Ref to [#29547](https://github.com/pandas-dev/pandas/issues/29547)

- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Replacing strings interpolated with .format() with strings interpolated with f-strings

### CODE REVIEW

File: pandas/tests/series/test_constructors.py [hunk 1]
```diff
@@ -1347,12 +1347,11 @@ def test_convert_non_ns(self):
...
 
         # floats are not ok
-        msg = "Cannot cast {}.*? to ".format(
-            # strip Index to convert PeriodIndex -> Period
-            # We don't care whether the error message says
-            # PeriodIndex or PeriodArray
-            type(index).__name__.rstrip("Index")
-        )
+        msg = f"Cannot cast {type(index).__name__.rstrip('Index')}.*? to "
+        # strip Index to convert PeriodIndex -> Period
```
Comment (WillAyd) [inline pandas/tests/series/test_constructors.py]: I guess should move this comment before 1350 so it reads sequentially

**Review by WillAyd — CHANGES_REQUESTED** [review]
very minor comment otherwise lgtm

File: pandas/tests/series/test_constructors.py (hunk 1 above)
Comment (monicaw218) [reply pandas/tests/series/test_constructors.py]: Thanks, I was wondering about this, too

**Review by WillAyd — APPROVED** [review]
lgtm ping on green

**Comment by WillAyd** [comment]
Great thanks @monicaw218

**Pull request merged**